apps/wallet/src/ui/app/components/transactions-card/Transaction.tsx

```diff
@@ -0,0 +1,199 @@
+    );
+
+    const amount = useMemo(() => {
+        const amount = amountByRecipient && amountByRecipient[0]?.amount;
+        const amountTransfers = eventsSummary.reduce(
+            (acc, { amount }) => acc + amount,
+            0
+        );
+
+        return Math.abs(amount || amountTransfers);
+    }, [amountByRecipient, eventsSummary]);
+
+    const recipientAddress = useMemo(() => {
+        const tranferObjectRecipientAddress =
+            amountByRecipient &&
+            amountByRecipient?.find(
+                ({ recipientAddress }) => recipientAddress !== address
+            )?.recipientAddress;
+        const receiverAddr =
+            eventsSummary &&
+            eventsSummary.find(
+                ({ receiverAddress }) => receiverAddress !== address
+            )?.receiverAddress;
+
+        return (
+            receiverAddr ??
+            tranferObjectRecipientAddress ??
+            certificate.data.sender
+        );
+    }, [address, amountByRecipient, certificate.data.sender, eventsSummary]);
+
+    // receiving transaction from move contract sometimes has the sender as the receiver
+    // so we need to check if the sender is the same as the address
+    const isSender = address === certificate.data.sender;
+
+    const receiverAddress = useMiddleEllipsis(
+        recipientAddress || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+
+    const txnLabel = useMemo(() => {
+        const moveCallTxn = getMoveCallTransaction(
+            certificate.data.transactions[0]
+        );
+        if (txnKind === 'Call')
+            return moveCallTxn?.function.replace(/_/g, ' ') || 'Call';
+        if (txnKind === 'ChangeEpoch') return txnKind;
+        return recipientAddress;
+    }, [certificate.data.transactions, recipientAddress, txnKind]);
+
+    const txnDate = useMemo(() => {
+        return txn?.timestamp_ms
+            ? formatDate(txn.timestamp_ms, ['month', 'day', 'hour', 'minute'])
+            : false;
+    }, [txn]);
+
+    const error = useMemo(() => getExecutionStatusError(txn), [txn]);
+    const isSuTransfer =
+        txnKind === 'PaySui' ||
+        txnKind === 'TransferSui' ||
+        txnKind === 'PayAllSui' ||
+        txnKind === 'Pay';
+
+    const label = useMemo(() => {
+        return isSuTransfer || txnKind === 'TransferObject'
+            ? isSender
+                ? 'To'
+                : 'From'
+            : 'Action';
+    }, [isSender, isSuTransfer, txnKind]);
+
+    return (
+        <Link
+            to={`/receipt?${new URLSearchParams({
+                txdigest: txn.certificate.transactionDigest,
+            }).toString()}`}
+            className="flex items-center w-full flex-col gap-2 py-4 no-underline"
+        >
+            <div className="flex items-start w-full justify-between gap-3">
+                <div className="w-7.5">
+                    <TxnIcon
+                        txnFailed={executionStatus === 'Failed' || !!error}
```

Comment:
I don't think `executionStatus` can ever `=== 'Failed'` based on the return value of `getExecutionStatus`

Reply:
removed it